# Task 2 — Sales Data Analysis

## Objective

The objective of this task is to analyze sales transaction data to evaluate business performance, identify sales and customer patterns, and generate evidence-based business insights.

The analysis focuses on core sales KPIs, time-based trends, category and product performance, regional differences, and customer purchasing behavior.

## Dataset Overview

The analysis uses the Sample Superstore sales dataset.

The dataset contains **9,994 rows** representing individual order line items and includes information about orders, customers, products, sales, quantities, discounts, profits, geographic locations, and shipping.

Initial inspection identified:

- **9,994 line-item records**
- **5,009 unique orders**
- **793 unique customers**
- **1,850 unique products**

Because a single order can contain multiple products, `order_id` may appear on multiple rows. Therefore, the number of orders is calculated using the number of unique order IDs rather than the number of dataset rows.

In [ ]:
# Loading the dataset
import pandas as pd

df = pd.read_csv('C:\\Users\Suman\Desktop\Job\Assignment 1\Sample - Superstore.csv', encoding="cp1252")

In [4]:
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/08/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/08/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,06/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [6]:
# Summary statistics using .describe()
df.describe(include='all')

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
count,9994.000000,9994,9994,9994,9994,9994,9994,9994,9994,9994,...,9994.000000,9994,9994,9994,9994,9994,9994.000000,9994.000000,9994.000000,9994.000000
unique,NaN,5009,1237,1334,4,793,793,3,1,531,...,NaN,4,1862,3,17,1850,NaN,NaN,NaN,NaN
top,NaN,CA-2017-100111,09/05/2016,12/16/2015,Standard Class,WB-21850,William Brown,Consumer,United States,New York City,...,NaN,West,OFF-PA-10001970,Office Supplies,Binders,Staple envelope,NaN,NaN,NaN,NaN
freq,NaN,14,38,35,5968,37,37,5191,9994,915,...,NaN,3203,19,6026,1523,48,NaN,NaN,NaN,NaN
mean,4997.500000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,55190.379428,NaN,NaN,NaN,NaN,NaN,229.858001,3.789574,0.156203,28.656896
std,2885.163629,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,32063.693350,NaN,NaN,NaN,NaN,NaN,623.245101,2.225110,0.206452,234.260108
min,1.000000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,1040.000000,NaN,NaN,NaN,NaN,NaN,0.444000,1.000000,0.000000,-6599.978000
25%,2499.250000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,23223.000000,NaN,NaN,NaN,NaN,NaN,17.280000,2.000000,0.000000,1.728750
50%,4997.500000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,56430.500000,NaN,NaN,NaN,NaN,NaN,54.490000,3.000000,0.200000,8.666500
75%,7495.750000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,90008.000000,NaN,NaN,NaN,NaN,NaN,209.940000,5.000000,0.200000,29.364000


In [7]:
# Checking if missing data or null values are present in the dataset

df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

In [8]:
missing_report = pd.DataFrame({
    'missing_count': df.isnull().sum(),
    'missing_percentage': (df.isnull().mean() * 100).round(2)
})

missing_report[missing_report['missing_count'] > 0]

,missing_count,missing_percentage


In [9]:
# Renaming columns according to snake casing for better readability and documentation

df.columns = df.columns.str.lower()
df.columns = df.columns.str.replace(' ','_')


In [10]:
df.columns

Index(['row_id', 'order_id', 'order_date', 'ship_date', 'ship_mode',
       'customer_id', 'customer_name', 'segment', 'country', 'city', 'state',
       'postal_code', 'region', 'product_id', 'category', 'sub-category',
       'product_name', 'sales', 'quantity', 'discount', 'profit'],
      dtype='object')

In [13]:
df = df.rename(columns={'sub-category':'sub_category'})

In [14]:
print("Total rows:", len(df))
print("Unique orders:", df['order_id'].nunique())
print("Unique customers:", df['customer_id'].nunique())
print("Unique products:", df['product_name'].nunique())

Total rows: 9994
Unique orders: 5009
Unique customers: 793
Unique products: 1850


In [15]:
df.columns

Index(['row_id', 'order_id', 'order_date', 'ship_date', 'ship_mode',
       'customer_id', 'customer_name', 'segment', 'country', 'city', 'state',
       'postal_code', 'region', 'product_id', 'category', 'sub_category',
       'product_name', 'sales', 'quantity', 'discount', 'profit'],
      dtype='object')

In [16]:
print("Fully duplicated rows:", df.duplicated().sum())
print("Duplicate Row IDs:", df['row_id'].duplicated().sum())

Fully duplicated rows: 0
Duplicate Row IDs: 0


In [17]:
# Converting date columns to datetime format

df['order_date'] = pd.to_datetime(df['order_date'],format='%m/%d/%Y',errors='coerce')

df['ship_date'] = pd.to_datetime(df['ship_date'],format='%m/%d/%Y',errors='coerce')

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   row_id         9994 non-null   int64         
 1   order_id       9994 non-null   object        
 2   order_date     9994 non-null   datetime64[ns]
 3   ship_date      9994 non-null   datetime64[ns]
 4   ship_mode      9994 non-null   object        
 5   customer_id    9994 non-null   object        
 6   customer_name  9994 non-null   object        
 7   segment        9994 non-null   object        
 8   country        9994 non-null   object        
 9   city           9994 non-null   object        
 10  state          9994 non-null   object        
 11  postal_code    9994 non-null   int64         
 12  region         9994 non-null   object        
 13  product_id     9994 non-null   object        
 14  category       9994 non-null   object        
 15  sub_category   9994 n

In [19]:
print("Invalid order dates:", df['order_date'].isna().sum())
print("Invalid ship dates:", df['ship_date'].isna().sum())

Invalid order dates: 0
Invalid ship dates: 0


In [20]:
# Creating time-based columns for trend analysis

df['year'] = df['order_date'].dt.year
df['month'] = df['order_date'].dt.month_name()
df['year_month'] = df['order_date'].dt.to_period('M')

In [21]:
df[['order_date', 'year', 'month', 'year_month']].head()

,order_date,year,month,year_month
0,2016-11-08,2016,November,2016-11
1,2016-11-08,2016,November,2016-11
2,2016-06-12,2016,June,2016-06
3,2015-10-11,2015,October,2015-10
4,2015-10-11,2015,October,2015-10


## 1. Data Preparation and Validation

Before calculating business metrics, the dataset was inspected and prepared for analysis.

The following checks and transformations were performed:

- Column names were standardized to lowercase `snake_case`.
- The `sub-category` field was standardized to `sub_category`.
- Fully duplicated rows were checked and none were found.
- Duplicate `row_id` values were checked and none were found.
- `order_date` and `ship_date` were converted to datetime format.
- Date conversion was validated and no invalid order or shipping dates were detected.
- `year`, `month`, and `year_month` fields were created from `order_date` for time-based analysis.

Repeated `order_id` values were retained because the dataset is stored at the order-line level, meaning one valid order can contain multiple product rows.

## 2. Order-Level Analysis Logic

The source dataset contains one row per order line rather than one row per complete order. Therefore, an order-level summary was created by grouping records using `order_id`.

For each unique order, the analysis aggregates:

- Total order revenue
- Total units purchased
- Customer ID
- Order date

This distinction is important because metrics such as **Total Orders** and **Average Order Value** must be calculated at the unique-order level rather than by counting individual product rows.

In [22]:
# Creating an order-level summary

order_summary = (
    df.groupby('order_id')
      .agg(
          order_revenue=('sales', 'sum'),
          total_units=('quantity', 'sum'),
          customer_id=('customer_id', 'first'),
          order_date=('order_date', 'first')
      )
      .reset_index()
)

order_summary.head()

,order_id,order_revenue,total_units,customer_id,order_date
0,CA-2014-100006,377.970,3,DK-13375,2014-09-07
1,CA-2014-100090,699.192,9,EB-13705,2014-07-08
2,CA-2014-100293,91.056,6,NF-18475,2014-03-14
3,CA-2014-100328,3.928,1,JC-15340,2014-01-28
4,CA-2014-100363,21.376,5,JM-15655,2014-04-08


## 3. Core KPI Analysis

Six core KPIs were calculated to summarize overall sales performance.

### KPI Definitions

- **Total Revenue:** Sum of sales across all order line items.
- **Total Orders:** Number of unique `order_id` values.
- **Average Order Value (AOV):** Total Revenue divided by Total Orders.
- **Total Units Sold:** Sum of the `quantity` field.
- **Top-Selling Category:** Product category contributing the highest total revenue.
- **Top Revenue-Generating Product:** Individual product generating the highest total sales revenue.

In [27]:
total_revenue = df['sales'].sum()

print(f"Total Revenue: ${total_revenue:,.2f}")

Total Revenue: $2,297,200.86


In [28]:
total_orders = df['order_id'].nunique()

print(f"Total Orders: {total_orders:,}")

Total Orders: 5,009


In [29]:
average_order_value = total_revenue / total_orders

print(f"Average Order Value: ${average_order_value:,.2f}")

Average Order Value: $458.61


In [30]:
total_units_sold = df['quantity'].sum()

print(f"Total Units Sold: {total_units_sold:,}")

Total Units Sold: 37,873


In [31]:
category_sales = (df.groupby('category')['sales'].sum().sort_values(ascending=False))
category_sales

category
Technology         836154.0330
Furniture          741999.7953
Office Supplies    719047.0320
Name: sales, dtype: float64

In [32]:
top_category = category_sales.idxmax()
top_category_revenue = category_sales.max()

print("Top_Selling_Category:", top_category)
print(f"Category_Revenue: ${top_category_revenue:,.2f}")

Top_Selling_Category: Technology
Category_Revenue: $836,154.03


In [35]:
product_sales = (df.groupby('product_name')['sales'].sum().sort_values(ascending=False))
product_sales.head(10)

product_name
Canon imageCLASS 2200 Advanced Copier                                          61599.824
Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind    27453.384
Cisco TelePresence System EX90 Videoconferencing Unit                          22638.480
HON 5400 Series Task Chairs for Big and Tall                                   21870.576
GBC DocuBind TL300 Electric Binding System                                     19823.479
GBC Ibimaster 500 Manual ProClick Binding System                               19024.500
Hewlett Packard LaserJet 3310 Copier                                           18839.686
HP Designjet T520 Inkjet Large Format Printer - 24" Color                      18374.895
GBC DocuBind P400 Electric Binding System                                      17965.068
High Speed Automatic Electric Letter Opener                                    17030.312
Name: sales, dtype: float64

In [36]:
top_product = product_sales.idxmax()
top_product_revenue = product_sales.max()

print("Top Revenue-Generating Product:", top_product)
print(f"Product Revenue: ${top_product_revenue:,.2f}")

Top Revenue-Generating Product: Canon imageCLASS 2200 Advanced Copier
Product Revenue: $61,599.82


In [ ]:
#KPI summary Tbale-
kpi_summary = pd.DataFrame({
    'KPI': [
        'Total Revenue',
        'Total Orders',
        'Average Order Value',
        'Total Units Sold',
        'Top-Selling Category',
        'Top Revenue-Generating Product'
    ],
    'Value': [
        f"${total_revenue:,.2f}",
        f"{total_orders:,}",
        f"${average_order_value:,.2f}",
        f"{total_units_sold:,}",
        top_category,
        top_product
    ]
})

kpi_summary

,KPI,Value
0,Total Revenue,"$2,297,200.86"
1,Total Orders,"5,009"
2,Average Order Value,$458.61
3,Total Units Sold,"37,873"
4,Top-Selling Category,Technology
5,Top Revenue-Generating Product,Canon imageCLASS 2200 Advanced Copier


### KPI Summary

The analysis produced the following results:

| KPI | Result |
|---|---:|
| Total Revenue | **$2,297,200.86** |
| Total Orders | **5,009** |
| Average Order Value | **$458.61** |
| Total Units Sold | **37,873** |
| Top-Selling Category | **Technology** |
| Top Revenue-Generating Product | **Canon imageCLASS 2200 Advanced Copier** |

Technology generated approximately **$836,154.03** in revenue, while the Canon imageCLASS 2200 Advanced Copier generated approximately **$61,599.82**, making them the leading category and product by revenue respectively.

## 4. Sales Trend Analysis

Sales performance was analyzed over time using yearly and monthly aggregations of revenue, unique orders, and units sold.

### Yearly Performance

Revenue changed as follows:

- **2014:** $484,247.50
- **2015:** $470,532.51
- **2016:** $609,205.60
- **2017:** $733,215.26

Revenue decreased by approximately **2.83% in 2015**, followed by strong increases of approximately **29.47% in 2016** and **20.36% in 2017**.

The number of unique orders also increased from **969 in 2014** to **1,687 in 2017**, while units sold increased from **7,581** to **12,476**.

### Trend Interpretation

The results show that sales performance was not consistently increasing throughout the entire period because revenue declined slightly in 2015. However, the subsequent growth in both 2016 and 2017, together with increases in order count and units sold, indicates substantially stronger sales activity during the later years of the dataset.

The dataset alone does not identify the cause of this growth, so factors such as marketing activity, market expansion, or seasonal demand should not be presented as confirmed explanations.

In [39]:
yearly_sales = (
    df.groupby('year')
      .agg(
          total_revenue=('sales', 'sum'),
          total_orders=('order_id', 'nunique'),
          total_units=('quantity', 'sum')
      )
      .reset_index()
)

yearly_sales['revenue_growth_pct'] = (
    yearly_sales['total_revenue']
    .pct_change()
    .mul(100)
    .round(2)
)

yearly_sales

,year,total_revenue,total_orders,total_units,revenue_growth_pct
0,2014,484247.4981,969,7581,NaN
1,2015,470532.5090,1038,7979,-2.83
2,2016,609205.5980,1315,9837,29.47
3,2017,733215.2552,1687,12476,20.36


In [40]:
monthly_sales = (
    df.groupby('year_month')
      .agg(
          total_revenue=('sales', 'sum'),
          total_orders=('order_id', 'nunique'),
          total_units=('quantity', 'sum')
      )
      .reset_index()
)

monthly_sales.head()

,year_month,total_revenue,total_orders,total_units
0,2014-01,14236.895,32,284
1,2014-02,4519.892,28,159
2,2014-03,55691.009,71,585
3,2014-04,28295.345,66,536
4,2014-05,23648.287,69,466


In [41]:
monthly_sales['revenue_growth_pct'] = (
    monthly_sales['total_revenue']
    .pct_change()
    .mul(100)
    .round(2)
)

monthly_sales.head(12)

,year_month,total_revenue,total_orders,total_units,revenue_growth_pct
0,2014-01,14236.8950,32,284,NaN
1,2014-02,4519.8920,28,159,-68.25
2,2014-03,55691.0090,71,585,1132.13
3,2014-04,28295.3450,66,536,-49.19
4,2014-05,23648.2870,69,466,-16.42
5,2014-06,34595.1276,66,521,46.29
6,2014-07,33946.3930,65,550,-1.88
7,2014-08,27909.4685,72,609,-17.78
8,2014-09,81777.3508,130,1000,193.01
9,2014-10,31453.3930,78,573,-61.54


In [43]:
highest_sales_month = monthly_sales.loc[
    monthly_sales['total_revenue'].idxmax()
]

highest_sales_month

year_month               2017-11
total_revenue         118447.825
total_orders                 261
total_units                 1840
revenue_growth_pct         52.29
Name: 46, dtype: object

In [44]:
lowest_sales_month = monthly_sales.loc[
    monthly_sales['total_revenue'].idxmin()
]

lowest_sales_month

year_month             2014-02
total_revenue         4519.892
total_orders                28
total_units                159
revenue_growth_pct      -68.25
Name: 1, dtype: object

### Monthly Sales Performance

The highest-revenue month in the dataset was **November 2017**, generating approximately **$118,447.82** from **261 orders** and **1,840 units sold**.

The lowest-revenue month was **February 2014**, with approximately **$4,519.89** in revenue from **28 orders** and **159 units sold**.

These results demonstrate substantial variation in monthly sales performance. November 2017 represents the strongest observed monthly performance, but the dataset alone is insufficient to attribute the peak to a specific seasonal, promotional, or external cause.

In [45]:
monthly_pattern = (
    df.groupby(df['order_date'].dt.month)
      .agg(
          total_revenue=('sales', 'sum'),
          total_orders=('order_id', 'nunique'),
          total_units=('quantity', 'sum')
      )
      .reset_index()
)

monthly_pattern['month_name'] = pd.to_datetime(
    monthly_pattern['order_date'],
    format='%m'
).dt.month_name()

monthly_pattern

,order_date,total_revenue,total_orders,total_units,month_name
0,1,94924.8356,178,1475,January
1,2,59751.2514,162,1067,February
2,3,205005.4888,354,2564,March
3,4,137762.1286,343,2447,April
4,5,155028.8117,369,2791,May
5,6,152718.6793,364,2680,June
6,7,147238.0970,338,2705,July
7,8,159044.0630,341,2784,August
8,9,307649.9457,688,5062,September
9,10,200322.9847,417,3104,October


## 5. Category and Product Performance

Product performance was evaluated using both revenue and unit volume.

### Category Performance

Technology generated the highest category revenue at approximately **$836,154.03**, representing **36.40%** of total revenue.

Furniture generated approximately **$741,999.80** (**32.30%**), while Office Supplies generated approximately **$719,047.03** (**31.30%**).

Although Technology was the largest revenue contributor, revenue was distributed relatively closely across all three major product categories.

### Product Performance

The **Canon imageCLASS 2200 Advanced Copier** was the highest-revenue product, generating approximately **$61,599.82** from **20 units sold**.

In contrast, **Staples** was the highest-volume product with **215 units sold**.

Therefore, the highest-revenue product and highest-volume product were not the same. This illustrates why both revenue and unit volume should be considered when evaluating product performance.

In [46]:
category_performance = (
    df.groupby('category')
      .agg(
          total_revenue=('sales', 'sum'),
          total_orders=('order_id', 'nunique'),
          total_units=('quantity', 'sum'),
          total_profit=('profit', 'sum')
      )
      .sort_values('total_revenue', ascending=False)
)

category_performance['revenue_share_pct'] = (
    category_performance['total_revenue']
    / total_revenue * 100
).round(2)

category_performance

,total_revenue,total_orders,total_units,total_profit,revenue_share_pct
category,,,,,
Technology,836154.0330,1544,6939,145454.9481,36.4
Furniture,741999.7953,1764,8028,18451.2728,32.3
Office Supplies,719047.0320,3742,22906,122490.8008,31.3


In [47]:
product_performance = (
    df.groupby('product_name')
      .agg(
          total_revenue=('sales', 'sum'),
          total_units=('quantity', 'sum'),
          total_orders=('order_id', 'nunique'),
          total_profit=('profit', 'sum')
      )
      .sort_values('total_revenue', ascending=False)
)

product_performance.head(10)

,total_revenue,total_units,total_orders,total_profit
product_name,,,,
Canon imageCLASS 2200 Advanced Copier,61599.824,20,5,2.519993e+04
Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind,27453.384,31,10,7.753039e+03
Cisco TelePresence System EX90 Videoconferencing Unit,22638.480,6,1,-1.811078e+03
HON 5400 Series Task Chairs for Big and Tall,21870.576,39,8,5.684342e-14
GBC DocuBind TL300 Electric Binding System,19823.479,37,11,2.233505e+03
GBC Ibimaster 500 Manual ProClick Binding System,19024.500,48,9,7.609800e+02
Hewlett Packard LaserJet 3310 Copier,18839.686,38,8,6.983884e+03
"HP Designjet T520 Inkjet Large Format Printer - 24"" Color",18374.895,12,3,4.094977e+03
GBC DocuBind P400 Electric Binding System,17965.068,27,6,-1.878166e+03


In [48]:
top_products_by_units = (
    df.groupby('product_name')
      .agg(
          total_units=('quantity', 'sum'),
          total_revenue=('sales', 'sum')
      )
      .sort_values(
          ['total_units', 'total_revenue'],
          ascending=False
      )
)

top_products_by_units.head(10)

,total_units,total_revenue
product_name,,
Staples,215,755.470
Staple envelope,170,1686.812
Easy-staple paper,150,2504.192
Staples in misc. colors,86,478.812
KI Adjustable-Height Table,74,4552.641
Storex Dura Pro Binders,71,278.586
Avery Non-Stick Binders,71,217.316
GBC Premium Transparent Covers with Diagonal Lined Pattern,67,780.456
"Situations Contoured Folding Chairs, 4/Set",64,3641.274


In [49]:
category_yearly_sales = (
    df.pivot_table(
        index='year',
        columns='category',
        values='sales',
        aggfunc='sum'
    )
    .round(2)
)

category_yearly_sales

category,Furniture,Office Supplies,Technology
year,,,
2014,157192.85,151776.41,175278.23
2015,170518.24,137233.46,162780.81
2016,198901.44,183939.98,226364.18
2017,215387.27,246097.18,271730.81


## 6. Regional Performance

Sales performance was compared across geographic regions using revenue, unique orders, customers, units sold, profit, and average order value.

The regional revenue ranking was:

1. **West:** $725,457.82 — 31.58% of total revenue
2. **East:** $678,781.24 — 29.55%
3. **Central:** $501,239.89 — 21.82%
4. **South:** $391,721.90 — 17.05%

The **West** was therefore the highest-revenue region, generating approximately **$725,457.82** from **1,611 unique orders**.

However, the East recorded a higher average order value than the West. This demonstrates that the region generating the most total revenue does not necessarily have the highest average value per order.

In [50]:
region_performance = (
    df.groupby('region')
      .agg(
          total_revenue=('sales', 'sum'),
          total_orders=('order_id', 'nunique'),
          unique_customers=('customer_id', 'nunique'),
          total_units=('quantity', 'sum'),
          total_profit=('profit', 'sum')
      )
      .sort_values('total_revenue', ascending=False)
)

region_performance['average_order_value'] = (
    region_performance['total_revenue']
    / region_performance['total_orders']
)

region_performance['revenue_share_pct'] = (
    region_performance['total_revenue']
    / total_revenue * 100
).round(2)

region_performance.round(2)

,total_revenue,total_orders,unique_customers,total_units,total_profit,average_order_value,revenue_share_pct
region,,,,,,,
West,725457.82,1611,686,12266,108418.45,450.32,31.58
East,678781.24,1401,674,10618,91522.78,484.50,29.55
Central,501239.89,1175,629,8780,39706.36,426.59,21.82
South,391721.90,822,512,6209,46749.43,476.55,17.05


In [51]:
state_performance = (
    df.groupby('state')
      .agg(
          total_revenue=('sales', 'sum'),
          total_orders=('order_id', 'nunique'),
          total_units=('quantity', 'sum'),
          total_profit=('profit', 'sum')
      )
      .sort_values('total_revenue', ascending=False)
)

state_performance.head(10).round(2)

,total_revenue,total_orders,total_units,total_profit
state,,,,
California,457687.63,1021,7667,76381.39
New York,310876.27,562,4224,74038.55
Texas,170188.05,487,3724,-25729.36
Washington,138641.27,256,1883,33402.65
Pennsylvania,116511.91,288,2153,-15559.96
Florida,89473.71,200,1379,-3399.30
Illinois,80166.10,276,1845,-12607.89
Ohio,78258.14,236,1759,-16971.38
Michigan,76269.61,117,946,24463.19


In [52]:
city_performance = (
    df.groupby(['state', 'city'])
      .agg(
          total_revenue=('sales', 'sum'),
          total_orders=('order_id', 'nunique'),
          unique_customers=('customer_id', 'nunique'),
          total_units=('quantity', 'sum')
      )
      .sort_values('total_revenue', ascending=False)
)

city_performance.head(10).round(2)

,,total_revenue,total_orders,unique_customers,total_units
state,city,,,,
New York,New York City,256368.16,450,355,3417
California,Los Angeles,175851.34,384,304,2879
Washington,Seattle,119540.74,212,188,1590
California,San Francisco,112669.09,265,232,1935
Pennsylvania,Philadelphia,109077.01,265,239,1981
Texas,Houston,64504.76,188,172,1466
Illinois,Chicago,48539.54,171,155,1132
California,San Diego,47521.03,88,86,670
Michigan,Detroit,42446.94,53,50,441


In [53]:
top_region = region_performance.index[0]

print("Top Revenue-Generating Region:", top_region)
print(
    f"Revenue: ${region_performance.loc[top_region, 'total_revenue']:,.2f}"
)

Top Revenue-Generating Region: West
Revenue: $725,457.82


## 7. Customer Insights

Customer-level performance was analyzed using revenue, unique order count, units purchased, profit, and average order value.

The dataset contains **793 unique customers**.

### Highest-Value and Most Frequent Customers

**Sean Miller** generated the highest total customer revenue:

- Revenue: **$25,043.05**
- Orders: **5**

The most frequent customer was **Emily Phan**:

- Orders: **17**
- Revenue: **$5,478.06**

The highest-revenue customer was therefore not the customer who placed the most orders. This suggests that customer value should be evaluated using both purchase frequency and monetary contribution rather than order frequency alone.

In [54]:
customer_performance = (
    df.groupby('customer_id')
      .agg(
          customer_name=('customer_name', 'first'),
          total_revenue=('sales', 'sum'),
          total_orders=('order_id', 'nunique'),
          total_units=('quantity', 'sum'),
          total_profit=('profit', 'sum')
      )
)

customer_performance['average_order_value'] = (
    customer_performance['total_revenue']
    / customer_performance['total_orders']
)

customer_performance.head()

,customer_name,total_revenue,total_orders,total_units,total_profit,average_order_value
customer_id,,,,,,
AA-10315,Alex Avila,5563.560,5,30,-362.8825,1112.712000
AA-10375,Allen Armold,1056.390,9,41,277.3824,117.376667
AA-10480,Andrew Allen,1790.512,4,36,435.8274,447.628000
AA-10645,Anna Andreadi,5086.935,6,64,857.8033,847.822500
AB-10015,Aaron Bergman,886.156,3,13,129.3465,295.385333


In [55]:
top_customers_by_revenue = (
    customer_performance
    .sort_values('total_revenue', ascending=False)
)

top_customers_by_revenue.head(10).round(2)

,customer_name,total_revenue,total_orders,total_units,total_profit,average_order_value
customer_id,,,,,,
SM-20320,Sean Miller,25043.05,5,50,-1980.74,5008.61
TC-20980,Tamara Chand,19052.22,5,42,8981.32,3810.44
RB-19360,Raymond Buch,15117.34,6,71,6976.10,2519.56
TA-21385,Tom Ashbrook,14595.62,4,36,4703.79,3648.90
AB-10105,Adrian Barton,14473.57,10,73,5444.81,1447.36
KL-16645,Ken Lonsdale,14175.23,12,113,806.86,1181.27
SC-20095,Sanjit Chand,14142.33,9,87,5757.41,1571.37
HL-15040,Hunter Lopez,12873.30,6,50,5622.43,2145.55
SE-20110,Sanjit Engle,12209.44,11,78,2650.68,1109.95


In [56]:
top_customers_by_orders = (
    customer_performance
    .sort_values(
        ['total_orders', 'total_revenue'],
        ascending=False
    )
)

top_customers_by_orders.head(10).round(2)

,customer_name,total_revenue,total_orders,total_units,total_profit,average_order_value
customer_id,,,,,,
EP-13915,Emily Phan,5478.06,17,124,144.96,322.24
ZC-21910,Zuschuss Carroll,8025.71,13,105,-1032.15,617.36
JE-15745,Joel Eaton,6760.82,13,90,221.80,520.06
SH-19975,Sally Hughsby,3406.84,13,113,558.47,262.06
CK-12205,Chloris Kastensmidt,3154.86,13,122,141.28,242.68
PG-18820,Patrick Gardner,3086.91,13,83,137.46,237.45
NS-18640,Noel Staavos,2964.82,13,113,-234.77,228.06
EA-14035,Erin Ashbrook,2846.70,13,87,-52.74,218.98
KL-16645,Ken Lonsdale,14175.23,12,113,806.86,1181.27


In [57]:
repeat_customers = customer_performance[
    customer_performance['total_orders'] > 1
]

one_time_customers = customer_performance[
    customer_performance['total_orders'] == 1
]

print("Repeat customers:", len(repeat_customers))
print("One-time customers:", len(one_time_customers))

Repeat customers: 781
One-time customers: 12


In [58]:
repeat_customer_pct = (
    len(repeat_customers)
    / len(customer_performance)
    * 100
)

print(f"Repeat Customer Percentage: {repeat_customer_pct:.2f}%")

Repeat Customer Percentage: 98.49%


In [59]:
repeat_customer_revenue = repeat_customers['total_revenue'].sum()

repeat_customer_revenue_share = (
    repeat_customer_revenue
    / total_revenue
    * 100
)

print(f"Repeat Customer Revenue: ${repeat_customer_revenue:,.2f}")
print(
    f"Repeat Customer Revenue Share: "
    f"{repeat_customer_revenue_share:.2f}%"
)

Repeat Customer Revenue: $2,292,033.22
Repeat Customer Revenue Share: 99.78%


### Repeat Customer Analysis

Customers with more than one unique order were classified as repeat customers.

The analysis identified:

- Total customers: **793**
- Repeat customers: **781**
- One-time customers: **12**
- Repeat customer rate: **98.49%**
- Repeat customer revenue share: **99.78%**

Repeat customers account for nearly all revenue observed in this dataset. This shows that the historical sales represented in the dataset are overwhelmingly associated with customers who placed multiple orders.

This result describes the purchasing pattern in the available dataset and should not, by itself, be interpreted as evidence that repeat purchasing caused higher revenue.

In [60]:
segment_performance = (
    df.groupby('segment')
      .agg(
          total_revenue=('sales', 'sum'),
          total_orders=('order_id', 'nunique'),
          unique_customers=('customer_id', 'nunique'),
          total_units=('quantity', 'sum'),
          total_profit=('profit', 'sum')
      )
      .sort_values('total_revenue', ascending=False)
)

segment_performance['average_order_value'] = (
    segment_performance['total_revenue']
    / segment_performance['total_orders']
)

segment_performance['revenue_share_pct'] = (
    segment_performance['total_revenue']
    / total_revenue
    * 100
).round(2)

segment_performance.round(2)

,total_revenue,total_orders,unique_customers,total_units,total_profit,average_order_value,revenue_share_pct
segment,,,,,,,
Consumer,1161401.34,2586,409,19521,134119.21,449.11,50.56
Corporate,706146.37,1514,236,11608,91979.13,466.41,30.74
Home Office,429653.15,909,148,6744,60298.68,472.67,18.70


### Customer Segment Performance

The Consumer segment generated the largest share of revenue:

- **Consumer:** $1,161,401.34 — 50.56%
- **Corporate:** $706,146.37 — 30.74%
- **Home Office:** $429,653.15 — 18.70%

Consumer was also the largest segment by unique customers and order count, with **409 customers** and **2,586 unique orders**.

The results indicate that the Consumer segment represents the largest portion of the observed sales activity and revenue within this dataset.

## 8. Business Questions

The completed analysis was used to answer a set of practical business questions covering sales trends, product performance, regional contribution, customer segments, repeat purchasing, and customer value.

The following questions are evaluated using the calculated metrics:

1. Which month recorded the highest sales?
2. Which category contributed the most revenue?
3. Which product generated the most revenue?
4. Was the highest-revenue product also the highest-volume product?
5. Which region generated the most revenue?
6. Which customer segment generated the most revenue?
7. How important were repeat customers to observed revenue?
8. Who were the highest-value and most frequent customers?

In [ ]:
#1. Which month recorded the highest sales?
highest_month = monthly_sales.loc[
    monthly_sales['total_revenue'].idxmax()
]

print("Highest Sales Month:", highest_month['year_month'])
print(f"Revenue: ${highest_month['total_revenue']:,.2f}")
print(f"Orders: {highest_month['total_orders']:,}")
print(f"Units Sold: {highest_month['total_units']:,}")

Highest Sales Month: 2017-11
Revenue: $118,447.82
Orders: 261
Units Sold: 1,840


In [ ]:
#2. Which category contributes the most revenue?
top_category_data = category_performance.iloc[0]

print("Top Revenue Category:", category_performance.index[0])
print(f"Revenue: ${top_category_data['total_revenue']:,.2f}")
print(
    f"Revenue Share: "
    f"{top_category_data['revenue_share_pct']:.2f}%"
)

Top Revenue Category: Technology
Revenue: $836,154.03
Revenue Share: 36.40%


In [ ]:
#3. Which product generates the most revenue?
top_product_data = product_performance.iloc[0]

print(
    "Top Revenue-Generating Product:",
    product_performance.index[0]
)

print(
    f"Revenue: "
    f"${top_product_data['total_revenue']:,.2f}"
)

print(
    f"Units Sold: "
    f"{top_product_data['total_units']:,.0f}"
)

Top Revenue-Generating Product: Canon imageCLASS 2200 Advanced Copier
Revenue: $61,599.82
Units Sold: 20


In [ ]:
#4. Is the highest-revenue product also the highest-volume product?
highest_revenue_product = product_performance.index[0]
highest_volume_product = top_products_by_units.index[0]

print("Highest Revenue Product:", highest_revenue_product)
print("Highest Volume Product:", highest_volume_product)

print(
    "Units sold for highest-volume product:",
    top_products_by_units.iloc[0]['total_units']
)

Highest Revenue Product: Canon imageCLASS 2200 Advanced Copier
Highest Volume Product: Staples
Units sold for highest-volume product: 215.0


In [ ]:
#5. Which region contributes the most revenue?
top_region_data = region_performance.iloc[0]

print("Top Revenue Region:", region_performance.index[0])
print(f"Revenue: ${top_region_data['total_revenue']:,.2f}")
print(
    f"Revenue Share: "
    f"{top_region_data['revenue_share_pct']:.2f}%"
)

print(
    f"Total Orders: "
    f"{top_region_data['total_orders']:,.0f}"
)

Top Revenue Region: West
Revenue: $725,457.82
Revenue Share: 31.58%
Total Orders: 1,611


In [ ]:
#6. Which customer segment contributes the most revenue?
top_segment_data = segment_performance.iloc[0]

print("Top Customer Segment:", segment_performance.index[0])
print(f"Revenue: ${top_segment_data['total_revenue']:,.2f}")
print(
    f"Revenue Share: "
    f"{top_segment_data['revenue_share_pct']:.2f}%"
)

print(
    f"Unique Customers: "
    f"{top_segment_data['unique_customers']:,.0f}"
)

Top Customer Segment: Consumer
Revenue: $1,161,401.34
Revenue Share: 50.56%
Unique Customers: 409


In [ ]:
#7. How important are repeat customers?
print(f"Total Customers: {len(customer_performance):,}")
print(f"Repeat Customers: {len(repeat_customers):,}")
print(f"One-Time Customers: {len(one_time_customers):,}")

print(
    f"Repeat Customer Percentage: "
    f"{repeat_customer_pct:.2f}%"
)

print(
    f"Repeat Customer Revenue Share: "
    f"{repeat_customer_revenue_share:.2f}%"
)

Total Customers: 793
Repeat Customers: 781
One-Time Customers: 12
Repeat Customer Percentage: 98.49%
Repeat Customer Revenue Share: 99.78%


In [ ]:
#8. Who are the highest-value and most frequent customers?
highest_value_customer = top_customers_by_revenue.iloc[0]
most_frequent_customer = top_customers_by_orders.iloc[0]

print("Highest Revenue Customer:")
print(highest_value_customer['customer_name'])
print(
    f"Revenue: "
    f"${highest_value_customer['total_revenue']:,.2f}"
)
print(
    f"Orders: "
    f"{highest_value_customer['total_orders']:,.0f}"
)

print("\nMost Frequent Customer:")
print(most_frequent_customer['customer_name'])
print(
    f"Orders: "
    f"{most_frequent_customer['total_orders']:,.0f}"
)
print(
    f"Revenue: "
    f"${most_frequent_customer['total_revenue']:,.2f}"
)

Highest Revenue Customer:
Sean Miller
Revenue: $25,043.05
Orders: 5

Most Frequent Customer:
Emily Phan
Orders: 17
Revenue: $5,478.06


In [72]:
#To summarise all-
analysis_summary = pd.DataFrame({
    'Business Question': [
        'Highest sales month',
        'Top revenue category',
        'Top revenue product',
        'Top volume product',
        'Top revenue region',
        'Top customer segment',
        'Repeat customer revenue share',
        'Most frequent customer'
    ],
    'Result': [
        str(highest_month['year_month']),
        category_performance.index[0],
        product_performance.index[0],
        top_products_by_units.index[0],
        region_performance.index[0],
        segment_performance.index[0],
        f"{repeat_customer_revenue_share:.2f}%",
        most_frequent_customer['customer_name']
    ]
})

analysis_summary

,Business Question,Result
0,Highest sales month,2017-11
1,Top revenue category,Technology
2,Top revenue product,Canon imageCLASS 2200 Advanced Copier
3,Top volume product,Staples
4,Top revenue region,West
5,Top customer segment,Consumer
6,Repeat customer revenue share,99.78%
7,Most frequent customer,Emily Phan


## 9. Key Observations

Based on the completed analysis, the following evidence-based observations were identified:

1. **Overall sales volume was substantial:** The dataset contains **5,009 unique orders**, generating approximately **$2.30 million in revenue** from **37,873 units sold**, with an average order value of **$458.61**.

2. **Sales activity strengthened in the later years:** Revenue declined slightly by **2.83% in 2015**, but subsequently increased by **29.47% in 2016** and **20.36% in 2017**. Both order count and units sold also increased over this period.

3. **November 2017 recorded the highest monthly revenue:** The month generated approximately **$118,447.82** from **261 orders** and **1,840 units**, making it the strongest sales month observed in the dataset.

4. **Technology was the leading category by revenue:** It generated approximately **$836,154.03**, representing **36.40%** of total revenue. However, Furniture and Office Supplies also contributed substantial shares, indicating that revenue was not dominated by a single category.

5. **Revenue leadership and volume leadership differed at the product level:** The Canon imageCLASS 2200 Advanced Copier generated the highest product revenue at approximately **$61,599.82**, while Staples recorded the highest unit volume at **215 units**.

6. **The West generated the most regional revenue:** It contributed approximately **$725,457.82**, or **31.58%** of total revenue, from **1,611 unique orders**.

7. **Repeat customers dominated the observed customer base and revenue:** **781 of 793 customers (98.49%)** placed more than one order, and these customers generated approximately **99.78%** of total revenue.

8. **Customer frequency and customer value were not equivalent:** Sean Miller generated the highest customer revenue at approximately **$25,043.05** from five orders, while Emily Phan placed the most orders (**17**) but generated approximately **$5,478.06** in revenue.

## 10. Analysis Limitations and Assumptions

The following limitations and analytical assumptions should be considered when interpreting the results:

- The dataset is stored at the **order-line level**, so unique `order_id` values are used when calculating order-based metrics.
- Revenue is calculated from the provided `sales` field.
- The analysis describes patterns contained in the dataset and does not establish causal explanations for changes in sales.
- Monthly and yearly differences may reflect multiple factors that are not represented in the available variables.
- Repeat customers are defined as customers with more than one unique order in the dataset.
- The analysis is limited to the time period and customers represented in the supplied Sample Superstore dataset.

These assumptions ensure that observations remain supported by the available data and are not interpreted beyond what the dataset can demonstrate.